packages/agreement-process/src/model/domain/agreement-validators.ts

```diff
@@ -273,6 +273,38 @@ export const assertRequesterIsDelegateConsumer = (
   }
 };
 
+export const assertRequesterisDelegateConsumerOrConsumer = async (
```

Comment:
Just a suggestion: this is a particular validator just for that operation, I would use a more specific name, something like `assertRequesterCanRequestAgrementForTenant`, or `assertRequesterCanCreateAgrementForTenant`

```suggestion
export const assertRequesterCanRequestAgrementForTenant = async (
```

Reply:
I opted for `assertRequesterCanCreateAgrementForTenant` since is more specific